### PR TITLE
[module] Add claim_shield module

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

We'll need `m_IsClaimable` later down the line for things like DI etc., this just adds it and uses it for a lua module claimshield. Seems to work nicely. Avoids any core changes

### CODE REVIEW

File: modules/custom/claim_shield.lua [hunk 1]
```diff
@@ -0,0 +1,93 @@
+-----------------------------------
+-- Claim Shield
+-----------------------------------
+require("modules/module_utils")
+require("scripts/globals/msg")
+require("scripts/globals/utils")
+-----------------------------------
+local m = Module:new("claim_shield")
+m:setEnabled(true)
```
Comment (claywar) [inline modules/custom/claim_shield.lua]: Should we default to enabled=false for custom modules?

File: src/map/lua/lua_baseentity.cpp [hunk 2]
```diff
@@ -9819,6 +9822,39 @@ sol::table CLuaBaseEntity::getNotorietyList()
...
+/************************************************************************
+ *  Function: setClaimable(...)
+ *  Purpose : sets m_IsClaimable for a mob
+ *  Example : mob:setClaimable(false)
+ *  Notes   :
+ ************************************************************************/
+
+void CLuaBaseEntity::setClaimable(bool claimable)
+{
+    XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB);
```
Comment (claywar) [inline src/map/lua/lua_baseentity.cpp]: Do we want to continue to use breaks in these bindings, or return gracefully and log?

**Review by claywar — APPROVED** [review]

File: modules/custom/claim_shield.lua (hunk 1 above)
Comment (zach2good) [reply modules/custom/claim_shield.lua]: Whoops, yes

File: src/map/lua/lua_baseentity.cpp (hunk 2 above)
Comment (zach2good) [reply src/map/lua/lua_baseentity.cpp]: Yeah, agreed

**Pull request merged**